modules/update-duration-cvo.adoc

```diff
@@ -12,3 +12,27 @@ The Cluster Version Operator (CVO) retrieves the target update release image and
 ====
 The CVO phase of the update does not restart the nodes.
 ====
+
+[discrete]
+[id="update-duration-example_{context}"]
+== Example update duration of cluster Operators
+
+// image to be replaced with a properly designed version
+image::update-duration.png[A diagram displaying the periods during which cluster Operators report as progressing during an update]
+
+The previous diagram shows an example of the time that cluster Operators might take to update to their new versions.
+The bars on the graph represent periods during which the respective Operator reports a `Progressing=True` condition.
+During an update, the `Progressing=True` condition is triggered mostly when the Operator is updating itself, but for some Operators, other tasks related to the update may trigger the condition as well.
+
+Each cluster Operator has characteristics that affect the time it takes to update itself.
+For instance, the `kube-apiserver` Operator in this example took over nine minutes to update because it was serving customer requests and is designed to be conservative about draining out existing requests before shutting down a pod.
+In the case of the `kube-apiserver` Operator, update speed is sacrificed to help prevent and limit disruptions to cluster functionality during an update.
+
+Another characteristic that affects the update duration of an Operator is whether the Operator utilizes DaemonSets.
+The `network` and `dns` Operators utilize full-cluster DaemonSets, which can take time to roll, and this is one of several reasons why these Operators might take longer to update themselves.
```

Comment:
@wking could you clarify what action is being taken with the DaemonSets during an update? You mentioned "roll" in the Slack thread but I'm not clear on what that means.

Reply:
These DaemonSets currently use `maxUnavailable: 10%` or 33% or `maxSurge: 10%` or similar.  That means that when the incoming component operator (e.g. the DNS operator) notices that the DNS DaemonSet needs a bump (e.g. to roll out the incoming CoreDNS container image), 10% of the pods will be replaced at a time.  So if you have 10 or more nodes where the DNS DaemonSet is running, that's 10 or more replacement pods coming up and replaced pods shutting down (with replacement-scale-up first in the `maxSurge` case and outgoing-shut-down first in the `maxUnavailable` case).  Those pods don't usually take too long to roll per-node, but it doesn't take much delay per-node before doing 10 of those in series gets you up into the minutes timescale.